src/beast/beast/unit_test/reporter.h

```diff
@@ -25,178 +25,313 @@
+    on_log (std::string const& s) override;
+};
+
+//------------------------------------------------------------------------------
+
+template <class _>
+reporter<_>::case_results::case_results (
+        std::string const& name_)
+    : name (name_)
+{
+}
+
+template <class _>
+reporter<_>::suite_results::suite_results (
+        std::string const& name_)
+    : name (name_)
+    , start (clock_type::now())
+{
+}
+
+template <class _>
+void
+reporter<_>::suite_results::add (case_results const& r)
+{
+    ++cases;
+    total += r.total;
+    failed += r.failed;
+}
+
+template <class _>
+reporter<_>::results::results()
+    : start (clock_type::now())
+{
+}
+
+template <class _>
+void
+reporter<_>::results::add (
+    suite_results const& r)
+{
+    ++suites;
+    total += r.total;
+    cases += r.cases;
+    failed += r.failed;
+
+    auto const elapsed =
+        clock_type::now() - r.start;
+    if (elapsed >= std::chrono::seconds(1))
     {
-        ++m_case.failed;
-        ++m_case.total;
-
-        m_stream.get() <<
-            "#" << m_case.total <<
-            " failed" <<
-            (reason.empty() ? "" : ": ") << reason
-            ;
+        auto const iter = std::lower_bound(top.begin(),
+            top.end(), elapsed,
+            [](run_time const& t1,
+                typename clock_type::duration const& t2)
+            {
+                return t1.second > t2;
+            });
+        if (iter != top.end())
+        {
+            if (top.size() == max_top)
+                top.resize(top.size() - 1);
+            top.emplace(iter, r.name, elapsed);
+        }
+        else if (top.size() < max_top)
+        {
+            top.emplace_back(r.name, elapsed);
+        }
     }
+}
 
-    virtual
-    void
-    on_log (std::string const& s) override
+//------------------------------------------------------------------------------
+
+template <class _>
+reporter<_>::reporter (
+        std::ostream& stream)
+    : std_ostream_ (std::ref (stream))
+    , stream_ (*std_ostream_)
+{
+}
+
+template <class _>
+reporter<_>::~reporter()
+{
+    if (results_.top.size() > 0)
     {
-        m_stream.get() <<
-            s;
+        stream_.get() << "Longest suite times:";
+        for (auto const& i : results_.top)
+            stream_.get() << std::setw(8) <<
+                fmtdur(i.second) << " " << i.first;
     }
-};
+    auto const elapsed =
+        clock_type::now() - results_.start;
+    stream_.get() <<
+        fmtdur(elapsed) << ", " <<
+        amount (results_.suites, "suite") << ", " <<
+        amount (results_.cases, "case") << ", " <<
+        amount (results_.total, "test") << " total, " <<
+        amount (results_.failed, "failure");
+}
+
+template <class _>
+reporter<_>::reporter (
+        abstract_ostream& stream)
+    : stream_ (stream)
+{
+}
+
+template <class _>
+std::string
+reporter<_>::fmtdur (
+    typename clock_type::duration const& d)
+{
+    using namespace std::chrono;
+    auto const ms =
```

Comment:
gotta get that return
key fixed it
seems to put returns
into very short lines.